compiler-rt/lib/scudo/standalone/secondary.h

```diff
@@ -321,9 +322,10 @@ class MapAllocatorCache {
         }
         CachedBlock PrevEntry = Quarantine[QuarantinePos];
         Quarantine[QuarantinePos] = Entry;
-        if (OldestTime == 0)
-          OldestTime = Entry.Time;
         Entry = PrevEntry;
+        // Update the entry time to reflect the time that the
+        // quarantined memory is placed in the Entries array
+        Entry.Time = Time;
```

Comment:
I noticed that we can't just change the time. Suppose the entry has been released then now we just mark it as unreleased again. This will cause some issue in MTE.

Reply:
Right, I will add a conditional to account for that case in the revised pull request.